### PR TITLE
Add autokick/chip and max allowed speed to MoveIntent/Primitive

### LINKED ISSUE #1959: Restructure MoveTactic and MoveIntent to support AutoChip/Kick
### Description of the task
With the previous tactic as coroutine system, we had multiple MoveActions what handled autochipping and autokicking. Now that we have tactic fsm and need to migrate these actions to the new system, it's worth rethinking whether these actions should be separate tactics or integrated as extensions of the MoveTactic. Below is one such idea:
* oneof proto representing options for autochipping/autokicking
* use the same oneof proto in the moveintent to pass into the navigator
* construct the move primitive using the oneof proto

### Acceptance criteria

- [ ] Plan the migration of autochip/autokick actions to the tactic fsm framework
- [ ] implement any fsm, navigator, intent, and primitive changes, including in firmware

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* adds a way to set a top speed for the robots
* adds a way to chip or kick using the moveintent/primitive

### Testing Done
unit testing

### Resolved Issues
resolves #1959 
part of #1882

### Length Justification
hard to pull these changes apart

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
nice yeetage of code

File: src/software/ai/hl/stp/tactic/move/move_tactic.h [hunk 1]
```diff
@@ -25,15 +25,35 @@ class MoveTactic : public Tactic
...
      * @param final_speed The final speed the robot should have at the destination
+     * @param dribbler_mode The dribbler mode
+     * @param ball_collision_type how to navigate around the ball
+     * @param autochipkick The command to autochip or autokick
+     */
+    void updateControlParams(
+        Point destination, Angle final_orientation, double final_speed,
+        DribblerMode dribbler_mode                           = DribblerMode::OFF,
+        BallCollisionType ball_collision_type                = BallCollisionType::AVOID,
+        std::optional<TbotsProto::Autochipkick> autochipkick = std::nullopt);
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/move/move_tactic.h]: changed to union

File: src/shared/proto/primitive.proto [hunk 2]
```diff
@@ -107,16 +93,20 @@ message DirectControlPrimitive
...
+        float chip_distance_meters     = 5;
+        float autokick_speed_m_per_s   = 6;
+        float autochip_distance_meters = 7;
     }
     float dribbler_speed_rpm = 8;
 }
 
-message MovePositionParams
+// Defined separately so that the definition can be shared
+message Autochipkick
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/shared/proto/primitive.proto]: changed to AutoChipOrKick

File: src/software/proto/primitive/primitive_msg_factory.cpp [hunk 3]
```diff
@@ -172,3 +132,17 @@ double convertDribblerModeToDribblerSpeed(DribblerMode dribbler_mode)
...
+    MaxAllowedSpeedMode max_allowed_speed_mode)
+{
+    switch (max_allowed_speed_mode)
+    {
+        case MaxAllowedSpeedMode::PHYSICAL_LIMIT:
+            return ROBOT_MAX_SPEED_METERS_PER_SECOND;
+        case MaxAllowedSpeedMode::STOP_COMMAND:
+            return STOP_COMMAND_ROBOT_MAX_SPEED_METERS_PER_SECOND;
+        default:
+            return ROBOT_MAX_SPEED_METERS_PER_SECOND;
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/proto/primitive/primitive_msg_factory.cpp]: logged warning

File: src/shared/constants.h [hunk 4]
```diff
@@ -25,6 +25,9 @@ const double BALL_MAX_RADIUS_METERS = 0.0215;
 const double MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT = 0.2;
 // The mass of a standard golf ball, as defined by https://en.wikipedia.org/wiki/Golf_ball
 const double BALL_MASS_KG = 0.004593;
+// The max allowed speed of the robot when the stop command is issued, in metres per
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/shared/constants.h]: ok

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :+1:

**Review by jsn0716 — APPROVED** [review]
LGTM 👍

**Pull request merged**